src/mesido/esdl/esdl_heat_model.py

```diff
@@ -759,6 +792,10 @@ def convert_heat_pump(
         if len(asset.in_ports) == 1 and len(asset.out_ports) == 1:
             _, modifiers = self.convert_heat_source(asset)
             return AirWaterHeatPump, modifiers
+        # In this case we only have the secondary side ports, here we assume a air-water HP elec
+        if len(asset.in_ports) == 2 and len(asset.out_ports) == 1:
+            _, modifiers = self.convert_air_water_heat_pump_elec(asset)
+            return AirWaterHeatPumpElec, modifiers
```

Comment:
instead of the many duplications compared to the AirWaterHeatPump can we not extend the dict modifiers which we would otherwise already get for the airwaterheatpump (without the electricity)

Reply:
Maybe, but it is not easy, as for the modifiers we go into functions that have if statements depending on the asset. Hence, although not pretty this works for now, I think we should tackle this in a larger refactor of this part of the code